lib/iris/tests/unit/fileformats/netcdf/test_Saver__ugrid.py

```diff
@@ -0,0 +1,1171 @@
+        self.assertEqual(sorted(mesh_names), ["Mesh2d", "Mesh2d_0"])
+
+        # they use different dimensions
+        self.assertEqual(
+            vars_meshdim(vars, "node", mesh_name="Mesh2d"), "Mesh2d_nodes"
+        )
+        self.assertEqual(
+            vars_meshdim(vars, "face", mesh_name="Mesh2d"), "Mesh2d_faces"
+        )
+        self.assertEqual(
+            vars_meshdim(vars, "node", mesh_name="Mesh2d_0"), "Mesh2d_nodes_0"
+        )
+        self.assertEqual(
+            vars_meshdim(vars, "face", mesh_name="Mesh2d_0"), "Mesh2d_faces_0"
+        )
+
+        # there are exactly two data-variables with a 'mesh' property
+        mesh_datavars = vars_w_props(vars, mesh="*")
+        self.assertEqual(["a", "b"], list(mesh_datavars))
+
+        # the data variables reference the two separate meshes
+        a_props, b_props = vars["a"], vars["b"]
+        self.assertEqual(a_props["mesh"], "Mesh2d")
+        self.assertEqual(a_props["location"], "face")
+        self.assertEqual(b_props["mesh"], "Mesh2d_0")
+        self.assertEqual(b_props["location"], "face")
+
+        # the data variables map the appropriate node dimensions
+        self.assertEqual(a_props["_DIMS"], ["Mesh2d_faces"])
+        self.assertEqual(b_props["_DIMS"], ["Mesh2d_faces_0"])
+
+    def test_multi_cubes_different_mesh(self):
+        # Check that we can correctly distinguish 2 different meshes.
+        cube1 = make_cube(var_name="a")
+        cube2 = make_cube(var_name="b", mesh=make_mesh(n_faces=4))
+
+        # Save and snapshot the result
+        tempfile_path = self.check_save_cubes([cube1, cube2])
+        dims, vars = scan_dataset(tempfile_path)
+
+        # there are 2 meshes in the file
+        mesh_names = vars_meshnames(vars)
+        self.assertEqual(len(mesh_names), 2)
+
+        # there are two (data)variables with a 'mesh' property
+        mesh_datavars = vars_w_props(vars, mesh="*")
+        self.assertEqual(2, len(mesh_datavars))
+        self.assertEqual(["a", "b"], sorted(mesh_datavars.keys()))
+
+        # the main variables reference the correct meshes, and 'face' location
+        a_props, b_props = vars["a"], vars["b"]
+        mesh_a, loc_a = a_props["mesh"], a_props["location"]
+        mesh_b, loc_b = b_props["mesh"], b_props["location"]
+        self.assertNotEqual(mesh_a, mesh_b)
+        self.assertEqual(loc_a, "face")
+        self.assertEqual(loc_b, "face")
+
+    def test_nonmesh_dim(self):
+        # Check where the data variable has a 'normal' dim and a mesh dim.
+        cube = make_cube()
+        cube = add_height_dim(cube)
+
+        # Save and snapshot the result
+        tempfile_path = self.check_save_cubes(cube)
+        dims, vars = scan_dataset(tempfile_path)
+
+        # have just 1 mesh, including a face and node coordinates.
+        (mesh_name,) = vars_meshnames(vars)
+        face_dim = vars_meshdim(vars, "face", mesh_name)
+        _ = vars_meshdim(vars, "node", mesh_name)
```

Comment:
@pp-mo Is it worthwhile checking that `face_dim == vars_meshdim(vars, "node", mesh_name)` ?